### PR TITLE
chore(tests): updating recent tags test

### PR DESCRIPTION
**Pull request by bassrock**
## Summary

Put back some recent tags logic https://github.com/Pocket/pocket-ios/pull/646#discussion_r1175344748

## References 
* Links to docs, tickets, designs if available

## Implementation Details
* Overview of work that was implemented and changes made to the codebase

## Test Steps
* Write out what QA should do to verify this change works as expected and hasn't introduced regressions. Can mention specific OS versions, devices, areas of the app to test as needed.

## PR Checklist:
- [ ] Added Unit / UI tests
- [ ] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/PocketKit/Tags/TagsFilterView.swift [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct TagsFilterView: View {
                     TagsCell(tag: .notTagged, tagAction: tagAction)
                         .disabled(isEditing)
                     TagsSectionView(
-                        showRecentTags: false,
+                        showRecentTags: viewModel.showRecentTags,
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Tags/TagsFilterView.swift:37]: @cyndichin I had to change some code for `test_archiveView_usingRecentTagFilter_withPremiumUser_showFilteredItems` to pass. is this intended?

Comment (cyndichin) [reply PocketKit/Sources/PocketKit/Tags/TagsFilterView.swift:37]: Ah the original code should be `showRecentTags: !isEditing && !viewModel.recentTags.isEmpty,`

File: PocketKit/Sources/PocketKit/Tags/TagsFilterViewModel.swift [hunk 2]
```diff
@@ -25,6 +25,10 @@ class TagsFilterViewModel: ObservableObject {
         return recentTagsFactory.recentTags.sorted().compactMap { TagType.recent($0) }
     }
 
+    var showRecentTags: Bool {
+        return user.status == .premium
+    }
+
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/Tags/TagsFilterViewModel.swift:31]: remove?

Comment (bassrock) [reply PocketKit/Sources/PocketKit/Tags/TagsFilterViewModel.swift]: fixed!

**Pull request merged**